content/docs/nextjs/github-public-repo.md

```diff
@@ -0,0 +1,303 @@
+
+```tsx
+// YourLayout.ts
+import { TinacmsGitHubProvider } from 'react-tinacms-github'
+
+const enterEditMode = () => {
+  return fetch(`/api/preview`).then(() => {
+    window.location.href = window.location.pathname
+  })
+}
+
+const exitEditMode = () => {
+  return fetch(`/api/reset-preview`).then(() => {
+    window.location.reload()
+  })
+}
+
+const YourLayout = ({ error, pageProps, children }) => {
+  return (
+    <TinacmsGitHubProvider
+      clientId={process.env.GITHUB_CLIENT_ID || ''}
+      authCallbackRoute="/api/create-github-access-token"
+      enterEditMode={enterEditMode}
+      exitEditMode={exitEditMode}
+      error={pageProps.error}
+    >
+      {children}
+    </TinacmsGitHubProvider>
+  )
+}
+```
+
+## Entering / Exiting "edit-mode"
+
+Next, we will need a way to enter/exit mode from our site. Let's create an "Edit Link" button. Ours will take `isEditing` as a parameter.
+
+_We'll be using Next.js's [preview-mode](https://nextjs.org/docs/advanced-features/preview-mode) to set this `isEditing` value. We'll set that up later_
+
+```tsx
+//...EditLink.tsx
+import React from 'react'
+import { useGitHubEditing } from 'react-tinacms-github'
+
+export interface EditLinkProps {
+  isEditing: boolean
+}
+
+export const EditLink = ({ isEditing }: EditLinkProps) => {
+  const github = useGitHubEditing()
+
+  return (
+    <button onClick={isEditing ? github.exitEditMode : github.enterEditMode}>
+      {isEditing ? 'Exit Edit Mode' : 'Edit This Site'}
+    </button>
+  )
+}
+```
+
+## Auth Redirects
+
+We will also a page to redirect the user to while authenticating with GitHub.
+
+```tsx
+//pages/github/authorizing.tsx
+// Our GitHub app redirects back to this page with auth code
+import React from 'react'
+import { useGitHubAuthRedirect } from 'react-tinacms-github'
+
+export default function Authorizing() {
+  // Let the main app know, that we receieved an auth code from the GitHub redirect
+  useGitHubAuthRedirect()
+  return <h2>Authorizing with GitHub, Please wait...</h2>
+}
+```
+
+## GitHub Oauth App
+
+In GitHub, within your account Settings, click [Oauth Apps](https://github.com/settings/developers) under Developer Settings.
+
+click "New Oauth App".
+
+For the **Authorization callback URL**, enter the url for the "authorizing" page that you created above (e.g https://your-url/github/authorizing). Fill out the other fields with your custom values.
+_Note: If you are testing your app locally, you may need a separate development GitHub app (with a localhost redirect), and a production GitHub app._
+
+Don't forget to add the **Client ID** and **Client Secret** to your environment variables.
+
+## Loading content from GitHub
+
+Now that we have access to the user's auth token, we can load content from GitHub within `getStaticProps`.
+
+```ts
+//About template about.tsx
+
+import { getGitHubPreviewProps, GitHubPreviewProps } from 'next-tinacms-github'
```

Comment:
```suggestion
import { 
    getGitHubPreviewProps, 
    GitHubPreviewProps 
} from 'next-tinacms-github'
```

Reply:
Small thing but it tripped me up — I was trying to import this from 'react-tinacms-github' and didn't notice the different package name since it's almost out of view in the docs.

Reply:
The import's actually wrong too:
```suggestion
import { getGithubPreviewProps, parseMarkdown } from 'next-tinacms-github';
import { GetStaticProps } from 'next';
```